### PR TITLE
Use grpc client for fulcio

### LINKED ISSUE #65: Change Fulcio to call V2 API
**Description**

We'd like to deprecate the V1 API, so we should switch the call to `api/v1/signingCert` to `api/v2/signingCert`. You can optionally switch to calls over gRPC too, but if you want to not pull in gRPC, HTTP still works.

cc @loosebazooka

### PR DESCRIPTION
**Pull request by loosebazooka**
- FulcioGrpcClient -> FulcioClient
- Remove old FulcioClient
- Fix some test where I was checking nullable
  on an optional "oops face!"

Signed-off-by: Appu Goundan <appu@google.com>

fixes #65

### CODE REVIEW

File: src/main/java/dev/sigstore/fulcio/client/FulcioClient.java [hunk 1]
```diff
@@ -86,43 +89,57 @@ public FulcioClient build() {
    *
    * @param cr certificate request parameters
    * @return a {@link SigningCertificate} from fulcio
-   * @throws IOException if the http request fials
```
Comment (patflynn) [inline src/main/java/dev/sigstore/fulcio/client/FulcioClient.java:89]: typo "fials"

File: src/main/java/dev/sigstore/fulcio/client/FulcioClient.java [hunk 2]
```diff
@@ -86,43 +89,57 @@ public FulcioClient build() {
...
-    try (InputStream content = resp.getContent()) {
-      var signingCert =
-          SigningCertificate.newSigningCertificate(
-              CharStreams.toString(new InputStreamReader(content, resp.getContentCharset())),
-              sctHeader);
-      if (signingCert.getDetachedSct().isEmpty() && !signingCert.hasEmbeddedSct() && requireSct) {
-        throw new IOException(
-            "no signed certificate timestamps were found in response from Fulcio");
+      throws InterruptedException, CertificateException, IOException {
+    // TODO: we should reuse this message channel across various executions of this client and
```
Comment (patflynn) [inline src/main/java/dev/sigstore/fulcio/client/FulcioClient.java:95]: what's the impact of not doing this? should there be a bug?

File: src/main/java/dev/sigstore/fulcio/client/FulcioClient.java [hunk 3]
```diff
@@ -86,43 +89,57 @@ public FulcioClient build() {
...
+              .build();
+
+      var certs = client.createSigningCertificate(req);
+
+      if (certs.getCertificateCase()
+          == dev.sigstore.fulcio.v2.SigningCertificate.CertificateCase
+              .SIGNED_CERTIFICATE_DETACHED_SCT) {
+        if (certs.getSignedCertificateDetachedSct().getSignedCertificateTimestamp().isEmpty()
+            && requireSct) {
+          throw new IOException(
```
Comment (patflynn) [inline src/main/java/dev/sigstore/fulcio/client/FulcioClient.java:129]: IOException is for unexpected network/file system errors isn't it?  This looks like serialization worked fine but for some reason the entry doesn't match expectations.  Perhaps FulcioVerificationException or just RuntimeException?

**Review by patflynn — APPROVED** [review]

File: src/main/java/dev/sigstore/fulcio/client/FulcioClient.java (hunk 2 above)
Comment (loosebazooka) [reply src/main/java/dev/sigstore/fulcio/client/FulcioClient.java:95]: yeah I think we would need an autoclosable or something so users could do:
```
try (var connection = FulcioClient.newConnection()) {
  connection.signingCert()
  ...
  connection.differentSigningCert()
}
```
Right now it's a bit expensive, but I don't think dangerous.

Comment (loosebazooka) [reply src/main/java/dev/sigstore/fulcio/client/FulcioClient.java:95]: I don't actually know a reason why someone would want to keep the connection open for too long with fulcio. We probably only need one cert to sign a bunch of artifacts.

Comment (loosebazooka) [reply src/main/java/dev/sigstore/fulcio/client/FulcioClient.java]: updated the comment

File: src/main/java/dev/sigstore/fulcio/client/FulcioClient.java (hunk 3 above)
Comment (loosebazooka) [reply src/main/java/dev/sigstore/fulcio/client/FulcioClient.java]: switched to CertificateException

**Review by patflynn — APPROVED** [review]

**Pull request merged**